### PR TITLE
[mini-PR] Fix a couple of new warnings

### PR DESCRIPTION
**Pull request by lucafedeli88**

### CODE REVIEW

File: Source/Particles/ParticleCreation/FilterCreateTransformFromFAB.H [hunk 1]
```diff
@@ -99,15 +100,33 @@ Index filterCreateTransformFromFAB (DstTile& dst1, DstTile& dst2, const amrex::B
     // Currently all particles are created on nodes. This makes it useless
     // to use N>1 (for now).
             const Real x = xlo_global + j*dx;
```
Comment (NeilZaim) [inline Source/Particles/ParticleCreation/FilterCreateTransformFromFAB.H]:
```suggestion
            const int l = iv[2];
    
            // Currently all particles are created on nodes. This makes it useless
            // to use N>1 (for now).
            const Real x = xlo_global + j*dx;
```

File: Source/Particles/ParticleCreation/FilterCreateTransformFromFAB.H [hunk 2]
```diff
@@ -99,15 +100,33 @@ Index filterCreateTransformFromFAB (DstTile& dst1, DstTile& dst2, const amrex::B
...
-#if (AMREX_SPACEDIM == 3)
             const int l = iv[2];
             const Real y = ylo_global + k*dy;
             const Real z = zlo_global + l*dz;
```
Comment (NeilZaim) [inline Source/Particles/ParticleCreation/FilterCreateTransformFromFAB.H]:
```suggestion
            const Real y = ylo_global + k*dy;
            const Real z = zlo_global + l*dz;

```

File: Source/Particles/ParticleCreation/FilterCreateTransformFromFAB.H [hunk 3]
```diff
@@ -99,15 +100,33 @@ Index filterCreateTransformFromFAB (DstTile& dst1, DstTile& dst2, const amrex::B
...
+        {
+            const IntVect iv = box.atOffset(i);
+            const int j = iv[0];
+            const int k = iv[1];
+    // Currently all particles are created on nodes. This makes it useless
+    // to use N>1 (for now).
+            const Real x = xlo_global + j*dx;
             const int l = 0;
             const int y = 0;
             const Real z = ylo_global + k*dy;
```
Comment (NeilZaim) [inline Source/Particles/ParticleCreation/FilterCreateTransformFromFAB.H]:
```suggestion
            const IntVect iv = box.atOffset(i);
            const int j = iv[0];
            const int k = iv[1];
            const int l = 0;

            // Currently all particles are created on nodes. This makes it useless
            // to use N>1 (for now).
            const Real x = xlo_global + j*dx;
            const Real y = Real(0.);
            const Real z = ylo_global + k*dy;
```

**Review by NeilZaim — COMMENTED** [review]
Thanks a lot! See minor suggestions about reformatting a bit below.

File: Source/Particles/ParticleCreation/FilterCreateTransformFromFAB.H [hunk 4]
```diff
@@ -89,25 +89,47 @@ Index filterCreateTransformFromFAB (DstTile& dst1, DstTile& dst2, const amrex::B
     // For loop over all cells in the box. If mask is true in the given cell,
     // we create the particles in the cell and apply a transform function to the
     // created particles.
+#if (AMREX_SPACEDIM == 3)
```
Comment (ax3l) [inline Source/Particles/ParticleCreation/FilterCreateTransformFromFAB.H]: This looks a bit too verbose on first sight.
How about this:
```
constexpr int spacedim = AMREX_SPACEDIM;
```
then below...

File: Source/Particles/ParticleCreation/FilterCreateTransformFromFAB.H [hunk 5]
```diff
@@ -89,25 +89,47 @@ Index filterCreateTransformFromFAB (DstTile& dst1, DstTile& dst2, const amrex::B
...
         if (mask[i])
         {
             const IntVect iv = box.atOffset(i);
             const int j = iv[0];
             const int k = iv[1];
-    // Currently all particles are created on nodes. This makes it useless
-    // to use N>1 (for now).
-            const Real x = xlo_global + j*dx;
-#if (AMREX_SPACEDIM == 3)
             const int l = iv[2];
```
Comment (ax3l) [inline Source/Particles/ParticleCreation/FilterCreateTransformFromFAB.H]:
```suggestion
            const int l = (spacedim == 3) ? iv[2] : 0;
```
and ...

File: Source/Particles/ParticleCreation/FilterCreateTransformFromFAB.H [hunk 6]
```diff
@@ -89,25 +89,47 @@ Index filterCreateTransformFromFAB (DstTile& dst1, DstTile& dst2, const amrex::B
...
+
+            // Currently all particles are created on nodes. This makes it useless
+            // to use N>1 (for now).
+            const Real x = xlo_global + j*dx;
             const Real y = ylo_global + k*dy;
```
Comment (ax3l) [inline Source/Particles/ParticleCreation/FilterCreateTransformFromFAB.H]:
```suggestion
            const Real y = (spacedim == 3) ? ylo_global + k*dy : 0.0_rt;
```

File: Source/Particles/ParticleCreation/FilterCreateTransformFromFAB.H (hunk 4 above)
Comment (lucafedeli88) [reply Source/Particles/ParticleCreation/FilterCreateTransformFromFAB.H]: It's a very good solution indeed! I've just implemented it...

**Review by NeilZaim — APPROVED** [review]
Looks good, thanks.

File: Source/Particles/ParticleCreation/FilterCreateTransformFromFAB.H (hunk 4 above)
Comment (ax3l) [reply Source/Particles/ParticleCreation/FilterCreateTransformFromFAB.H]: Hurray, glad you like it :)

**Review by ax3l — APPROVED** [review]
Thank you! :+1: :sparkles:

**Pull request merged**